### PR TITLE
chore: script to isolate formatting changes between commits

### PR DESCRIPTION
**Pull request by sdstolworthy**
#### Description of changes

This script isolates formatting changes between two revisions. This is a helpful tool to reduce formatting noise in PRs. It will allow developers submit two PRs, the first with formatting changes, the second with functional changes in code.

#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/extract-formatting-changes.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import { exec, execSync, spawn } from 'child_process';
+import yargs from 'yargs';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as readline from 'readline';
+
+const getArgs = (latestRevision: string) => {
+  return yargs(process.argv.slice(2))
+    .usage('Usage: $0 -1 [rev1] -2 [rev2] -b [branch]')
```
Comment (sobolk) [inline scripts/extract-formatting-changes.ts:9]: should `-1` and '-2` be `s` and `h` ?

**Review by sobolk — APPROVED** [review]

**Review by goldbez — DISMISSED** [review]

File: scripts/extract-formatting-changes.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    execSync(`git branch ${newBranchName} ${baseBranch}`);
+  } catch (e) {
+    if (e.status === 128) {
+      throw new Error('Branch already exists. Please delete it first.');
+    }
+  }
+};
+
+const checkoutBranch = (branch: string) => {
+  execSync(`git checkout ${branch}`);
```
Comment (jhockett) [inline scripts/extract-formatting-changes.ts:77]:
```suggestion
  execSync(`git switch ${branch}`);
```

**Review by goldbez — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

File: scripts/extract-formatting-changes.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    .usage('Usage: $0 -s [rev1] -h [rev2] -b [branch]')
+    .nargs('s', 1)
+    .describe('s', 'Base revision')
+    .default('s', 'dev')
+    .nargs('h', 1)
+    .describe('h', 'Head revision')
+    .default('h', latestRevision)
+    .nargs('b', 1)
+    .default('b', '')
+    .describe('b', 'Branch name').argv;
```
Comment (sobolk) [inline scripts/extract-formatting-changes.ts:18]: I liked the verbose `base` `head` and `branch` alternatives :-(

**Review by sobolk — APPROVED** [review]

File: scripts/extract-formatting-changes.ts (hunk 3 above)
Comment (sdstolworthy) [reply scripts/extract-formatting-changes.ts:18]: I wasn't getting the expected behavior from the aliases, so I ended up removing them.

**Pull request merged**